src/sql/workbench/services/objectExplorer/browser/asyncServerTreeRenderer.ts

```diff
@@ -115,6 +115,11 @@ class ConnectionProfileTemplate extends Disposable {
 			label = localize('loading', "Loading...");
 		}
 
+		if (element.providerName === 'KUSTO') {
```

Comment:
We already call this up above when the view isn't the compact version. What scenario are you trying to fix here?

Reply:
This is for the connection dialog. _isCompact is true so the above code doesn't get executed. Should I try to merge it into the above code and exclude the 'isconnected' part?

Reply:
I don't actually know why we wouldn't just call that for everything. So if you wouldn't mind here's the changes I'd suggest :

1. Remove renderServerIcon from the above blocks and then remove your check for Kusto so all providers get the icon even when in "compact" mode
2. Rename _isCompact to _showStatusBadge since that's really what it's doing